支持spacemit k1芯片的dts/clk/reset/pinctrl/gpio/uart等基础功能和配置

kevin-zhm opened the pull request:
issue: https://github.com/RVCK-Project/rvck/issues/19

执行下面的命令编译内核，得到kernel Image：
cat arch/riscv/configs/k1_defconfig >> arch/riscv/configs/openeuler_defconfig
make ARCH=riscv CROSS_COMPILE=riscv64-unknown-linux-gnu- openeuler_defconfig
make ARCH=riscv CROSS_COMPILE=riscv64-unknown-linux-gnu- -j8

通过fastboot下载spl、u-boot-opensbi、kernel Image、dtb和rootfs.cpio.gz可以正常启动到linux shell，启动log如下：
Starting kernel ...

[    0.000000][    T0] Linux version 6.6.0+ (zhangmeng@snode5) (riscv64-unknown-linux-gnu-gcc (g09b62c20e09) 13.2.1 20240423, GNU ld (GNU Binutils) 2.42) RVCK-Project/rvck#14 SMP PREEMPT Wed Mar 19 14:20:56 CST 2025
[    0.000000][    T0] Machine model: Banana Pi BPI-F3
[    0.000000][    T0] SBI specification v1.0 detected
[    0.000000][    T0] SBI implementation ID=0x1 Version=0x10003
[    0.000000][    T0] SBI IPI extension detected
[    0.000000][    T0] SBI RFENCE extension detected
[    0.000000][    T0] efi: UEFI not found.
[    0.000000][    T0] OF: reserved mem: 0x0000000000000000..0x000000000007ffff (512 KiB) nomap non-reusable mmode_resv0@0
[    0.000000][    T0] OF: reserved mem: 0x000000007f000000..0x000000007fffffff (16384 KiB) map non-reusable framebuffer@7f000000
[    0.000000][    T0] cma: Reserved 32 MiB at 0x000000007aa00000 on node -1
[    0.000000][    T0] NUMA: No NUMA configuration found
[    0.000000][    T0] NUMA: Faking a node at [mem 0x0000000000000000-0x000000027fffffff]
[    0.000000][    T0] NUMA: NODE_DATA [mem 0x27ffc9d00-0x27ffcefff]
[    0.000000][    T0] Zone ranges:
[    0.000000][    T0]   DMA32    [mem 0x0000000000000000-0x00000000ffffffff]
[    0.000000][    T0]   Normal   [mem 0x0000000100000000-0x000000027fffffff]
[    0.000000][    T0] Movable zone start for each node
[    0.000000][    T0] Early memory node ranges
[    0.000000][    T0]   node   0: [mem 0x0000000000000000-0x000000000007ffff]
[    0.000000][    T0]   node   0: [mem 0x0000000000080000-0x000000007fffffff]
[    0.000000][    T0]   node   0: [mem 0x0000000100000000-0x000000027fffffff]
[    0.000000][    T0] Initmem setup node 0 [mem 0x0000000000000000-0x000000027fffffff]
[    0.000000][    T0] SBI HSM extension detected
[    0.000000][    T0] riscv: base ISA extensions acdfimv
[    0.000000][    T0] riscv: ELF capabilities acdfim
[    0.000000][    T0] percpu: cpu 0 has no node -1 or node-local memory
[    0.000000][    T0] percpu: Embedded 31 pages/cpu s86248 r8192 d32536 u126976
[    0.000000][    T0] pcpu-alloc: s86248 r8192 d32536 u126976 alloc=31*4096
[    0.000000][    T0] pcpu-alloc: [0] 0 [0] 1 [0] 2 [0] 3 [0] 4 [0] 5 [0] 6 [0] 7 
[    0.000000][    T0] Kernel command line: earlycon=sbi console=ttySP0,115200n8 loglevel=8 rdinit=/init
[    0.000000][    T0] Dentry cache hash table entries: 1048576 (order: 11, 8388608 bytes, linear)
[    0.000000][    T0] Inode-cache hash table entries: 524288 (order: 10, 4194304 bytes, linear)
[    0.000000][    T0] Fallback order for Node 0: 0 
[    0.000000][    T0] Built 1 zonelists, mobility grouping on.  Total pages: 2064384
[    0.000000][    T0] Policy zone: Normal
[    0.000000][    T0] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000][    T0] stackdepot: allocating hash table via alloc_large_system_hash
[    0.000000][    T0] stackdepot hash table entries: 524288 (order: 10, 4194304 bytes, linear)
[    0.000000][    T0] software IO TLB: area num 8.
[    0.000000][    T0] software IO TLB: mapped [mem 0x0000000076a00000-0x000000007aa00000] (64MB)
[    0.000000][    T0] Virtual kernel memory layout:
[    0.000000][    T0]       fixmap : 0xffffffc6fea00000 - 0xffffffc6ff000000   (6144 kB)
[    0.000000][    T0]       pci io : 0xffffffc6ff000000 - 0xffffffc700000000   (  16 MB)
[    0.000000][    T0]      vmemmap : 0xffffffc700000000 - 0xffffffc800000000   (4096 MB)
[    0.000000][    T0]      vmalloc : 0xffffffc800000000 - 0xffffffd800000000   (  64 GB)
[    0.000000][    T0]      modules : 0xffffffff0281c000 - 0xffffffff80000000   (2007 MB)
[    0.000000][    T0]       lowmem : 0xffffffd800000000 - 0xffffffda80000000   (  10 GB)
[    0.000000][    T0]       kernel : 0xffffffff80000000 - 0xffffffffffffffff   (2047 MB)
[    0.000000][    T0] Memory: 8063752K/8388608K available (15777K kernel code, 5941K rwdata, 14124K rodata, 710K init, 4505K bss, 292088K reserved, 32768K cma-reserved)
[    0.000000][    T0] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=8, Nodes=1
[    0.000000][    T0] trace event string verifier disabled
[    0.000000][    T0] rcu: Preemptible hierarchical RCU implementation.
[    0.000000][    T0] rcu: RCU restricting CPUs from NR_CPUS=512 to nr_cpu_ids=8.
[    0.000000][    T0] Trampoline variant of Tasks RCU enabled.
[    0.000000][    T0] Tracing variant of Tasks RCU enabled.
[    0.000000][    T0] rcu: RCU calculated value of scheduler-enlistment delay is 25 jiffies.
[    0.000000][    T0] rcu: Adjusting geometry for rcu_fanout_leaf=16, nr_cpu_ids=8
[    0.000000][    T0] NR_IRQS: 64, nr_irqs: 64, preallocated irqs: 0
[    0.000000][    T0] riscv-intc: 64 local interrupts mapped
[    0.000000][    T0] plic: interrupt-controller@e0000000: mapped 159 interrupts with 8 handlers for 16 contexts.
[    0.000000][    T0] riscv: providing IPIs using SBI IPI extension
[    0.000000][    T0] rcu: srcu_init: Setting srcu_struct sizes based on contention.
[    0.000000][    T0] clocksource: riscv_clocksource: mask: 0xffffffffffffffff max_cycles: 0x588fe9dc0, max_idle_ns: 440795202592 ns
[    0.000001][    T0] sched_clock: 64 bits at 24MHz, resolution 41ns, wraps every 4398046511097ns
[    0.000020][    T0] riscv-timer: Timer interrupt in S-mode is available via sstc extension
[    0.000945][    T0] Console: colour dummy device 80x25
[    0.001696][    T0] Calibrating delay loop (skipped), value calculated using timer frequency.. 48.00 BogoMIPS (lpj=96000)
[    0.001709][    T0] pid_max: default: 32768 minimum: 301
[    0.004908][    T0] LSM: initializing lsm=lockdown,capability,yama,selinux,integrity
[    0.006447][    T0] Yama: becoming mindful.
[    0.006477][    T0] SELinux:  Initializing.
[    0.010884][    T0] Mount-cache hash table entries: 16384 (order: 5, 131072 bytes, linear)
[    0.010925][    T0] Mountpoint-cache hash table entries: 16384 (order: 5, 131072 bytes, linear)
[    0.018163][    T1] RCU Tasks: Setting shift to 3 and lim to 1 rcu_task_cb_adjust=1 rcu_task_cpu_ids=8.
[    0.018318][    T1] RCU Tasks Trace: Setting shift to 3 and lim to 1 rcu_task_cb_adjust=1 rcu_task_cpu_ids=8.
[    0.018465][    T1] riscv: ELF compat mode unsupported
[    0.018509][    T1] ASID allocator using 16 bits (65536 entries)
[    0.018708][    T1] rcu: Hierarchical SRCU implementation.
[    0.018713][    T1] rcu: Max phase no-delay instances is 1000.
[    0.020938][    T1] EFI services will not be available.
[    0.022232][    T1] smp: Bringing up secondary CPUs ...
[    0.044284][    T0] cpu1: Ratio of byte access time to unaligned word access is 10.12, unaligned accesses are fast
[    0.068334][    T0] cpu2: Ratio of byte access time to unaligned word access is 10.12, unaligned accesses are fast
[    0.092379][    T0] cpu3: Ratio of byte access time to unaligned word access is 10.12, unaligned accesses are fast
[    0.116414][    T0] cpu4: Ratio of byte access time to unaligned word access is 10.52, unaligned accesses are fast
[    0.140449][    T0] cpu5: Ratio of byte access time to unaligned word access is 10.12, unaligned accesses are fast
[    0.164481][    T0] cpu6: Ratio of byte access time to unaligned word access is 10.12, unaligned accesses are fast
[    0.188511][    T0] cpu7: Ratio of byte access time to unaligned word access is 10.56, unaligned accesses are fast
[    0.188654][    T1] smp: Brought up 1 node, 8 CPUs
[    0.195161][    T1] devtmpfs: initialized
[    0.214277][    T1] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 7645041785100000 ns
[    0.214313][    T1] futex hash table entries: 2048 (order: 5, 131072 bytes, linear)
[    0.216642][    T1] pinctrl core: initialized pinctrl subsystem
[    0.221557][    T1] NET: Registered PF_NETLINK/PF_ROUTE protocol family
[    0.224060][    T1] DMA: preallocated 1024 KiB GFP_KERNEL pool for atomic allocations
[    0.225456][    T1] DMA: preallocated 1024 KiB GFP_KERNEL|GFP_DMA32 pool for atomic allocations
[    0.225648][    T1] audit: initializing netlink subsys (disabled)
[    0.226164][   T59] audit: type=2000 audit(0.224:1): state=initialized audit_enabled=0 res=1
[    0.227791][    T1] thermal_sys: Registered thermal governor 'fair_share'
[    0.227799][    T1] thermal_sys: Registered thermal governor 'step_wise'
[    0.227802][    T1] thermal_sys: Registered thermal governor 'user_space'
[    0.227805][    T1] thermal_sys: Registered thermal governor 'power_allocator'
[    0.227898][    T1] cpuidle: using governor menu
[    0.248846][    T1] cpu0: Ratio of byte access time to unaligned word access is 10.52, unaligned accesses are fast
[    0.256146][    T1] pinctrl-single d401e000.pinctrl: 148 pins, size 592
[    0.260188][    T1] HugeTLB: registered 1.00 GiB page size, pre-allocated 0 pages
[    0.260199][    T1] HugeTLB: 16380 KiB vmemmap can be freed for a 1.00 GiB page
[    0.260204][    T1] HugeTLB: registered 64.0 KiB page size, pre-allocated 0 pages
[    0.260209][    T1] HugeTLB: 0 KiB vmemmap can be freed for a 64.0 KiB page
[    0.260213][    T1] HugeTLB: registered 2.00 MiB page size, pre-allocated 0 pages
[    0.260216][    T1] HugeTLB: 28 KiB vmemmap can be freed for a 2.00 MiB page
[    0.267967][    T1] gpio gpiochip0: Static allocation of GPIO base is deprecated, use dynamic allocation.
[    0.269378][    T1] ACPI: Interpreter disabled.
[    0.270119][    T1] iommu: Default domain type: Translated
[    0.270125][    T1] iommu: DMA domain TLB invalidation policy: lazy mode
[    0.275032][    T1] SCSI subsystem initialized
[    0.275610][    T1] libata version 3.00 loaded.
[    0.276143][    T1] usbcore: registered new interface driver usbfs
[    0.276212][    T1] usbcore: registered new interface driver hub
[    0.276373][    T1] usbcore: registered new device driver usb
[    0.276646][    T1] mc: Linux media interface: v0.10
[    0.276790][    T1] pps_core: LinuxPPS API ver. 1 registered
[    0.276795][    T1] pps_core: Software ver. 5.3.6 - Copyright 2005-2007 Rodolfo Giometti <giometti@linux.it>
[    0.277740][    T1] EDAC MC: Ver: 3.0.0
[    0.279555][    T1] Advanced Linux Sound Architecture Driver Initialized.
[    0.281310][    T1] Bluetooth: Core ver 2.22
[    0.281426][    T1] NET: Registered PF_BLUETOOTH protocol family
[    0.281430][    T1] Bluetooth: HCI device and connection manager initialized
[    0.281460][    T1] Bluetooth: HCI socket layer initialized
[    0.281470][    T1] Bluetooth: L2CAP socket layer initialized
[    0.281530][    T1] Bluetooth: SCO socket layer initialized
[    0.281635][    T1] NetLabel: Initializing
[    0.281638][    T1] NetLabel:  domain hash size = 128
[    0.281641][    T1] NetLabel:  protocols = UNLABELED CIPSOv4 CALIPSO
[    0.281811][    T1] NetLabel:  unlabeled traffic allowed by default
[    0.282415][    T1] vgaarb: loaded
[    0.283422][    T1] clocksource: Switched to clocksource riscv_clocksource
[    0.287210][    T1] VFS: Disk quotas dquot_6.6.0
[    0.287704][    T1] VFS: Dquot-cache hash table entries: 512 (order 0, 4096 bytes)
[    0.288983][    T1] pnp: PnP ACPI: disabled
[    0.307779][    T1] NET: Registered PF_INET protocol family
[    0.308196][    T1] IP idents hash table entries: 131072 (order: 8, 1048576 bytes, linear)
[    0.317877][    T1] tcp_listen_portaddr_hash hash table entries: 4096 (order: 4, 65536 bytes, linear)
[    0.317952][    T1] Table-perturb hash table entries: 65536 (order: 6, 262144 bytes, linear)
[    0.317988][    T1] TCP established hash table entries: 65536 (order: 7, 524288 bytes, linear)
[    0.318317][    T1] TCP bind hash table entries: 65536 (order: 9, 2097152 bytes, linear)
[    0.321359][    T1] TCP: Hash tables configured (established 65536 bind 65536)
[    0.321862][    T1] UDP hash table entries: 4096 (order: 5, 131072 bytes, linear)
[    0.321957][    T1] UDP-Lite hash table entries: 4096 (order: 5, 131072 bytes, linear)
[    0.323042][    T1] NET: Registered PF_UNIX/PF_LOCAL protocol family
[    0.324636][    T1] RPC: Registered named UNIX socket transport module.
[    0.324645][    T1] RPC: Registered udp transport module.
[    0.324649][    T1] RPC: Registered tcp transport module.
[    0.324653][    T1] RPC: Registered tcp-with-tls transport module.
[    0.324656][    T1] RPC: Registered tcp NFSv4.1 backchannel transport module.
[    0.326027][    T1] PCI: CLS 0 bytes, default 64
[    0.326228][    T1] kvm [1]: hypervisor extension not available
[    0.326962][   T77] Trying to unpack rootfs image as initramfs...
[    0.840581][    T1] Initialise system trusted keyrings
[    0.840704][    T1] Key type blacklist registered
[    0.841395][    T1] workingset: timestamp_bits=39 max_order=21 bucket_order=0
[    0.841572][    T1] zbud: loaded
[    0.855825][    T1] NFS: Registering the id_resolver key type
[    0.855910][    T1] Key type id_resolver registered
[    0.855914][    T1] Key type id_legacy registered
[    0.856120][    T1] nfs4filelayout_init: NFSv4 File Layout Driver Registering...
[    0.856208][    T1] jffs2: version 2.2. (NAND) ?2001-2006 Red Hat, Inc.
[    0.857643][    T1] fuse: init (API version 7.39)
[    0.858745][    T1] 9p: Installing v9fs 9p2000 file system support
[    0.861496][    T1] integrity: Platform Keyring initialized
[    0.895255][    T1] jitterentropy: Initialization failed with host not compliant with requirements: 9
[    0.895285][    T1] NET: Registered PF_ALG protocol family
[    0.895332][    T1] Key type asymmetric registered
[    0.895336][    T1] Asymmetric key parser 'x509' registered
[    0.895911][    T1] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 243)
[    0.896372][    T1] io scheduler mq-deadline registered
[    0.896379][    T1] io scheduler kyber registered
[    0.896695][    T1] io scheduler bfq registered
[    0.905351][    T1] shpchp: Standard Hot Plug PCI Controller Driver version: 0.4
[    0.909994][    T1] Serial: 8250/16550 driver, 6 ports, IRQ sharing enabled
[    0.915928][    T1] d4017000.serial: ttySP0 at MMIO 0xd4017000 (irq = 14, base_baud = 921250) is a UART1
[    0.916118][    T1] printk: console [ttySP0] enabled
[    1.116282][   T77] Freeing initrd memory: 18844K
[    1.121187][    T1] SuperH (H)SCI(F) driver initialized
[    2.196050][    T1] loop: module loaded
[    2.277034][    T1] rdac: device handler registered
[    2.282515][    T1] hp_sw: device handler registered
[    2.287598][    T1] emc: device handler registered
[    2.292822][    T1] alua: device handler registered
[    2.303707][    T1] e1000e: Intel(R) PRO/1000 Network Driver
[    2.309520][    T1] e1000e: Copyright(c) 1999 - 2015 Intel Corporation.
[    2.318677][    T1] usbcore: registered new interface driver uas
[    2.324947][    T1] usbcore: registered new interface driver usb-storage
[    2.331832][    T1] usbcore: registered new interface driver ums-sddr09
[    2.338639][    T1] usbcore: registered new interface driver ums-sddr55
[    2.345852][    T1] mousedev: PS/2 mouse device common for all mice
[    2.353197][    T1] i2c_dev: i2c /dev entries driver
[    2.359383][    T1] Watchdog module: th1520-wdt loaded
[    2.364698][    T1] device-mapper: core: CONFIG_IMA_DISABLE_HTABLE is disabled. Duplicate IMA measurements will not be recorded in the IMA log.
[    2.378057][    T1] device-mapper: uevent: version 1.0.3
[    2.384104][    T1] device-mapper: ioctl: 4.48.0-ioctl (2023-03-01) initialised: dm-devel@redhat.com
[    2.393471][    T1] Bluetooth: HCI UART driver ver 2.3
[    2.398715][    T1] Bluetooth: HCI UART protocol H4 registered
[    2.405387][    T1] sdhci: Secure Digital Host Controller Interface driver
[    2.412423][    T1] sdhci: Copyright(c) Pierre Ossman
[    2.417735][    T1] sdhci-pltfm: SDHCI platform and OF driver helper
[    2.424835][    T1] hid: raw HID events driver (C) Jiri Kosina
[    2.431966][    T1] usbcore: registered new interface driver usbhid
[    2.438366][    T1] usbhid: USB HID core driver
[    2.443875][    T1] riscv-pmu-sbi: SBI PMU extension is available
[    2.450223][    T1] riscv-pmu-sbi: 16 firmware and 18 hardware counters
[    2.509111][    T1] NET: Registered PF_INET6 protocol family
[    2.518375][    T1] Segment Routing with IPv6
[    2.523038][    T1] In-situ OAM (IOAM) with IPv6
[    2.527890][    T1] NET: Registered PF_PACKET protocol family
[    2.533877][    T1] bridge: filtering via arp/ip/ip6tables is no longer available by default. Update your scripts to load br_netfilter if you need this.
[    2.548205][    T1] Bluetooth: RFCOMM TTY layer initialized
[    2.553933][    T1] Bluetooth: RFCOMM socket layer initialized
[    2.559972][    T1] Bluetooth: RFCOMM ver 1.11
[    2.564548][    T1] Bluetooth: BNEP (Ethernet Emulation) ver 1.3
[    2.570696][    T1] Bluetooth: BNEP socket layer initialized
[    2.576478][    T1] Bluetooth: HIDP (Human Interface Emulation) ver 1.2
[    2.583245][    T1] Bluetooth: HIDP socket layer initialized
[    2.589923][    T1] 9pnet: Installing 9P2000 support
[    2.595206][    T1] Key type dns_resolver registered
[    2.631604][    T1] registered taskstats version 1
[    2.640890][    T1] Loading compiled-in X.509 certificates
[    2.721176][    T1] Loaded X.509 cert 'Build time autogenerated kernel key: 5db3d1cce4ed0a877c6d182224c4b4a8e069d922'
[    2.756071][    T1] kmemleak: Kernel memory leak detector initialized (mem pool available: 15560)
[    2.756078][  T132] kmemleak: Automatic memory scanning thread started
[    2.784999][    T1] Key type encrypted registered
[    2.789937][    T1] ima: IMA RoT initializing tpm
[    2.794759][    T1] ima: No RoT found, activating RoT-bypass!
[    2.800713][    T1] ima: Allocated hash algorithm: sha256
[    2.806365][    T1] ima: No architecture policies found
[    2.811876][    T1] evm: Initialising EVM extended attributes:
[    2.817858][    T1] evm: security.selinux
[    2.821980][    T1] evm: security.SMACK64 (disabled)
[    2.827055][    T1] evm: security.SMACK64EXEC (disabled)
[    2.832493][    T1] evm: security.SMACK64TRANSMUTE (disabled)
[    2.838358][    T1] evm: security.SMACK64MMAP (disabled)
[    2.843772][    T1] evm: security.apparmor
[    2.847959][    T1] evm: security.ima
[    2.851693][    T1] evm: security.capability
[    2.856046][    T1] evm: HMAC attrs: 0x1
[    2.898761][  T142] DRBG: Continuing without Jitter RNG
[    3.582941][    T1] th1520 rpmsg: driver is registered.
[    3.591007][    T1] cfg80211: Loading compiled-in X.509 certificates for regulatory database
[    3.600760][    T1] Loaded X.509 cert 'sforshee: 00b28ddf47aef9cea7'
[    3.608198][    T1] Loaded X.509 cert 'wens: 61c038651aabdcf94bd0ac7ff06c7248db18c600'
[    3.616664][    T1] clk: Disabling unused clocks
[    3.621611][   T96] platform regulatory.0: Direct firmware load for regulatory.db failed with error -2
[    3.623699][    T1] ALSA device list:
[    3.631155][   T96] cfg80211: failed to load regulatory.db
[    3.634880][    T1]   No soundcards found.
[    3.635006][    T1] integrity: Unable to open file: /etc/keys/x509_ima.der (-2)
[    3.644739][    T1] integrity: Unable to open file: /etc/keys/x509_evm.der (-2)
[    3.652474][    T1] Freeing unused kernel image (initmem) memory: 704K
[    3.666626][    T1] Kernel memory protection not selected by kernel config.
[    3.673768][    T1] Run /init as init process
[    3.678218][    T1]   with arguments:
[    3.681967][    T1]     /init
[    3.685010][    T1]   with environment:
[    3.688931][    T1]     HOME=/
[    3.692060][    T1]     TERM=linux
run init for debuging based on ramdisk ...

~ # ls
bin      include  linuxrc  proc     share    usr
dev      init     man      root     sys      var
etc      lib      mnt      sbin     tmp

Commit:
riscv: k1x: add spacemit k1 soc support

community inclusion
category: feature
bugzilla: https://github.com/RVCK-Project/rvck-olk/issues/7
Reference: https://gitee.com/bianbu-linux/linux-6.6/commit/0cd0f4131d633657b8d01465615b291419fc592a
Reference: https://gitee.com/bianbu-linux/linux-6.6/commit/9885e32a7616a4726de02f0d9576e78909d1aa13
Reference: https://gitee.com/bianbu-linux/linux-6.6/commit/04c4a6375dcf191677ee337954c118db819f36cf

--------------------------------

K1 is the first SoC in the Spacemit series, which contain 8 X60 cores in
2 clusters, support RISC-V Vector v1.0, and expand some matrix computing.

Signed-off-by: zhangmeng <zhangmeng.kevin@linux.spacemit.com>

Commit:
riscv: k1x: dt-bindings: add clock dt-bindings for spacemit k1

community inclusion
category: feature
bugzilla: https://github.com/RVCK-Project/rvck-olk/issues/7
Reference: https://gitee.com/bianbu-linux/linux-6.6/commit/91f61242fd22268d978c26aae1fddebc6d13b3d8
Reference: https://gitee.com/bianbu-linux/linux-6.6/commit/a6786e2320a61017b556e50d2c347df59c72507f

--------------------------------

Add clock deinitions for each module on the clock tree, clock source
is generated from the 24MHz oscillator.

Signed-off-by: zhangmeng <zhangmeng.kevin@linux.spacemit.com>

Commit:
driver: k1x: clk: add clock driver for spacemit k1

community inclusion
category: feature
bugzilla: https://github.com/RVCK-Project/rvck-olk/issues/7
Reference: https://gitee.com/bianbu-linux/linux-6.6/commit/91f61242fd22268d978c26aae1fddebc6d13b3d8

--------------------------------

The clock tree of K1 SoC contains three main types of clock hardware
(PLL/DDN/MIX) and has control registers split into several multifunction
devices: APBS (PLLs), MPMU, APBC and APMU.

Some registers are shared with clk and reset, so, there is a globle
spinlock for access protection.

Signed-off-by: zhangmeng <zhangmeng.kevin@linux.spacemit.com>

Commit:
riscv: k1x: dts: add clock support for spacemit k1

community inclusion
category: feature
bugzilla: https://github.com/RVCK-Project/rvck-olk/issues/7

--------------------------------

Describe the PLL and system controllers that're capable of generating
clock signals in the devicetree.

Signed-off-by: zhangmeng <zhangmeng.kevin@linux.spacemit.com>

Commit:
riscv: k1x: dt-bindings: add reset dt-bindings for spacemit k1

community inclusion
category: feature
bugzilla: https://github.com/RVCK-Project/rvck-olk/issues/7
Reference: https://gitee.com/bianbu-linux/linux-6.6/commit/eb84ca6c7d41e7da20973e67a05b97d95035bd49

--------------------------------

Add reset ID for each module which contain a reset bit to release the
hardware circuit reset signal.

Signed-off-by: zhangmeng <zhangmeng.kevin@linux.spacemit.com>

Commit:
driver: k1x: add reset driver for spacemit k1

community inclusion
category: feature
bugzilla: https://github.com/RVCK-Project/rvck-olk/issues/7
Reference: https://gitee.com/bianbu-linux/linux-6.6/commit/eb84ca6c7d41e7da20973e67a05b97d95035bd49

--------------------------------

Usually, a module just use 1-bit to control the reset signal,
a data structure spacemit_reset_signal is defined to describe
the reset signal for each module.

Signed-off-by: zhangmeng <zhangmeng.kevin@linux.spacemit.com>

Commit:
riscv: k1x: dts: add reset support for spacemit k1

community inclusion
category: feature
bugzilla: https://github.com/RVCK-Project/rvck-olk/issues/7
Reference: https://gitee.com/bianbu-linux/linux-6.6/commit/eb84ca6c7d41e7da20973e67a05b97d95035bd49

--------------------------------

Describe reset controller which generate reset signals for eache
module in the devicetree.

Signed-off-by: zhangmeng <zhangmeng.kevin@linux.spacemit.com>

Commit:
riscv: k1x: dt-bindings: support pinctrl binding for spacemit k1x soc

community inclusion
category: feature
bugzilla: https://github.com/RVCK-Project/rvck-olk/issues/7
Reference: https://gitee.com/bianbu-linux/linux-6.6/commit/d02470cdc7aa8704193148c44de7f49869595a53
Reference: https://gitee.com/bianbu-linux/linux-6.6/commit/cad40852d3882331380ed7cad6c669942e365426

--------------------------------

Add pinctrl bindings for Spacemit k1 SoC sys pinctrl controller.

Signed-off-by: zhangmeng <zhangmeng.kevin@linux.spacemit.com>

Commit:
driver: k1x: add pinctrl driver for spacemit k1x

community inclusion
category: feature
bugzilla: https://github.com/RVCK-Project/rvck-olk/issues/7
Reference: https://gitee.com/bianbu-linux/linux-6.6/commit/cad40852d3882331380ed7cad6c669942e365426

--------------------------------

SpacemiT's K1 SoC has a pinctrl controller which use single register
to describe all functions, which include bias pull up/down(strong pull),
drive strength, schmitter trigger, slew rate, mux mode.

Signed-off-by: zhangmeng <zhangmeng.kevin@linux.spacemit.com>

Commit:
riscv: k1x: dts: add pinctrl support for spacemit k1x soc

community inclusion
category: feature
bugzilla: https://github.com/RVCK-Project/rvck-olk/issues/7
Reference: https://gitee.com/bianbu-linux/linux-6.6/commit/d02470cdc7aa8704193148c44de7f49869595a53

--------------------------------

Describe the pin controller and pin mux in devicetree.

Signed-off-by: zhangmeng <zhangmeng.kevin@linux.spacemit.com>

Commit:
driver: k1x: add gpio driver for spacemit k1

community inclusion
category: feature
bugzilla: https://github.com/RVCK-Project/rvck-olk/issues/7
Reference: https://gitee.com/bianbu-linux/linux-6.6/commit/eeeae787980ad56ee28e055efe8f0301c88a5a3c

--------------------------------

Implement GPIO functionality which capable of setting pin as
input, output. Also, each pin can be used as interrupt which
support rising, falling, or both edge type trigger.

Signed-off-by: zhangmeng <zhangmeng.kevin@linux.spacemit.com>

Commit:
riscv: k1x: dts: add gpio support spacemit k1x soc

community inclusion
category: feature
bugzilla: https://github.com/RVCK-Project/rvck-olk/issues/7
Reference: https://gitee.com/bianbu-linux/linux-6.6/commit/eeeae787980ad56ee28e055efe8f0301c88a5a3c

--------------------------------

Describe the gpio controller and define some gpio in devicetree.

Signed-off-by: zhangmeng <zhangmeng.kevin@linux.spacemit.com>

Commit:
driver: k1x: add uart serial driver support for spacemit k1x

community inclusion
category: feature
bugzilla: https://github.com/RVCK-Project/rvck-olk/issues/7
Reference: https://gitee.com/bianbu-linux/linux-6.6/commit/3bb9da2e70626cea480d4065b16e835d6aeeccfa

--------------------------------

The uart in k1 is a pxa likely module, the driver is not based on
8250, because the devices of /dev/ttyS* node has been registered
by 8250, so k1 uart driver register /dev/ttySP* nodes.

Signed-off-by: zhangmeng <zhangmeng.kevin@linux.spacemit.com>

Commit:
riscv: k1x: dts: add uart support for spacemit k1x soc

community inclusion
category: feature
bugzilla: https://github.com/RVCK-Project/rvck-olk/issues/7
Reference: https://gitee.com/bianbu-linux/linux-6.6/commit/1da7a402280fc17aaf877a494803aa49404992c9
Reference: https://gitee.com/bianbu-linux/linux-6.6/commit/8334e3d07192288d22bf0ee5e1a914243c87905c
Reference: https://gitee.com/bianbu-linux/linux-6.6/commit/92b25650c377d016dd16cb5fd25cf3d36aca95ac
Reference: https://gitee.com/bianbu-linux/linux-6.6/commit/5b42acf4cb3337a92a1bd680b7281ce350242f42

--------------------------------

Add uart controller description to devietree, k1 has 9 uart
controllers, which numbered 0, 2~8, and the No.1 of uart is
reserved for secure world.

Signed-off-by: zhangmeng <zhangmeng.kevin@linux.spacemit.com>

Commit:
riscv: k1x: config: enable clock driver for spacemit k1

community inclusion
category: feature
bugzilla: https://github.com/RVCK-Project/rvck-olk/issues/7

--------------------------------

Signed-off-by: zhangmeng <zhangmeng.kevin@linux.spacemit.com>

Commit:
riscv: k1x: config: enable reset driver for spacemit k1

community inclusion
category: feature
bugzilla: https://github.com/RVCK-Project/rvck-olk/issues/7

--------------------------------

Signed-off-by: zhangmeng <zhangmeng.kevin@linux.spacemit.com>

Commit:
rsicv: k1x: config: enable pinctrl driver for spacemit k1

community inclusion
category: feature
bugzilla: https://github.com/RVCK-Project/rvck-olk/issues/7

--------------------------------

Signed-off-by: zhangmeng <zhangmeng.kevin@linux.spacemit.com>

Commit:
riscv: k1x: config: enable gpio driver for spacemit k1

community inclusion
category: feature
bugzilla: https://github.com/RVCK-Project/rvck-olk/issues/7

--------------------------------

Signed-off-by: zhangmeng <zhangmeng.kevin@linux.spacemit.com>

Commit:
riscv: config: enable serial driver for spacemit k1

community inclusion
category: feature
bugzilla: https://github.com/RVCK-Project/rvck-olk/issues/7

--------------------------------

Signed-off-by: zhangmeng <zhangmeng.kevin@linux.spacemit.com>

oervci commented:
开始测试

oervci commented:
check patch done. log: https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/check-patch/3/consoleFull

oervci commented:
Kernel build success!

oervci commented:
kunit test done. log:https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/kunit-test/5/consoleFull

oervci commented:
Lava check fail! log: https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/rvck-lava-trigger/14/consoleFull

sterling-teng commented:
这一版补丁依旧未适配openeuler_defconfig，可能我没有描述清楚，我又回复了你上一版补丁，麻烦点进已经close的17号pr里面浏览下，补丁的标题也需要按照我在17号pr里的评论格式化一下。

简单的说，rvck-olk的编译配置文件只有一个，即openeuler_defconfig，除此之外，arch/riscv/config/目录下其它的任何xxx_config都不再进行后续维护，所以，你可以不用添加k1_defconfig，把相关的config项移植到openeuler_defconfig中并确保该配置编译出来的内核能在你的硬件上boot_up即可。

kevin-zhm commented:
我要直接更新openeuler_defconfig这个文件吗？我验证的时候的方法如下：

执行下面的命令编译内核，得到kernel Image：
cat arch/riscv/configs/k1_defconfig >> arch/riscv/configs/openeuler_defconfig
make ARCH=riscv CROSS_COMPILE=riscv64-unknown-linux-gnu- openeuler_defconfig
make ARCH=riscv CROSS_COMPILE=riscv64-unknown-linux-gnu- -j8

是把k1所需要配置添加到openeuler_defconfig里，编译出的内核，可以下再看看最上面的log
对于openeuler_defconfig这个文件，现在执行 make ARCH=riscv CROSS_COMPILE=riscv64-unknown-linux-gnu- openeuler_defconfig之后，生成的.config和openeuler_defconfig的差别比较大，我不是很清楚openeuler_defconfig的维护机制是怎样的。我添加k1的配置到openeuler_defconfig，采用的是 cat k1_defconfig >> openeuler_defconfig，编译出的内核Image是可以启动k1板子的。问题的关键是我应该怎样更新 openeuler_defconfig文件？

sterling-teng commented:
kevin-zhm ***@***.***> 于2025年3月20日周四 10:42写道：

> 我要直接更新openeuler_defconfig这个文件吗？我验证的时候的方法如下：
>
> 执行下面的命令编译内核，得到kernel Image：
> cat arch/riscv/configs/k1_defconfig >>
> arch/riscv/configs/openeuler_defconfig
> make ARCH=riscv CROSS_COMPILE=riscv64-unknown-linux-gnu-
> openeuler_defconfig
> make ARCH=riscv CROSS_COMPILE=riscv64-unknown-linux-gnu- -j8
>
> 是把k1所需要配置添加到openeuler_defconfig里，编译出的内核，可以下再看看最上面的log
> 对于openeuler_defconfig这个文件，现在执行 make ARCH=riscv
> CROSS_COMPILE=riscv64-unknown-linux-gnu-
> openeuler_defconfig之后，生成的.config和openeuler_defconfig的差别比较大，我不是很清楚openeuler_defconfig的维护机制是怎样的。我添加k1的配置到openeuler_defconfig，采用的是
> cat k1_defconfig >> openeuler_defconfig，编译出的内核Image是可以启动k1板子的。问题的关键是我应该怎样更新
> openeuler_defconfig文件？
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/RVCK-Project/rvck-olk/pull/18#issuecomment-2738973267>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AOILBDVE7XBGHPRD5BBWAYT2VITLBAVCNFSM6AAAAABZKAFA52VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMZYHE3TGMRWG4>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
> [image: kevin-zhm]*kevin-zhm* left a comment (RVCK-Project/rvck-olk#18)
> <https://github.com/RVCK-Project/rvck-olk/pull/18#issuecomment-2738973267>
>
> 我要直接更新openeuler_defconfig这个文件吗？我验证的时候的方法如下：
>
是的。

> 执行下面的命令编译内核，得到kernel Image：
> cat arch/riscv/configs/k1_defconfig >>
> arch/riscv/configs/openeuler_defconfig
> make ARCH=riscv CROSS_COMPILE=riscv64-unknown-linux-gnu-
> openeuler_defconfig
>
这里最好执行：
```
 make ARCH=riscv menuconfig
```
检查下的配置项是不是如预期一样打开了。

> make ARCH=riscv CROSS_COMPILE=riscv64-unknown-linux-gnu- -j8
>
> 是把k1所需要配置添加到openeuler_defconfig里，编译出的内核，可以下再看看最上面的log
> 对于openeuler_defconfig这个文件，现在执行 make ARCH=riscv
> CROSS_COMPILE=riscv64-unknown-linux-gnu-
> openeuler_defconfig之后，生成的.config和openeuler_defconfig的差别比较大，我不是很清楚openeuler_defconfig的维护机制是怎样的。我添加k1的配置到openeuler_defconfig，采用的是
> cat k1_defconfig >> openeuler_defconfig，编译出的内核Image是可以启动k1板子的。问题的关键是我应该怎样更新
> openeuler_defconfig文件？
>
可以尝试下
```
make save-oedefconfig
```

> —
> Reply to this email directly, view it on GitHub
> <https://github.com/RVCK-Project/rvck-olk/pull/18#issuecomment-2738973267>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AOILBDVE7XBGHPRD5BBWAYT2VITLBAVCNFSM6AAAAABZKAFA52VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMZYHE3TGMRWG4>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

Commit:
riscv: config: Enable Spacemit k1 support

community inclusion
category: feature
bugzilla: https://github.com/RVCK-Project/rvck-olk/issues/7

--------------------------------

Based on the current openeuler_defconfig for riscv, use the following
commands to generate the new openeuler_defconfig:

cp arch/riscv/configs/openeuler_defconfig .config
cat arch/riscv/configs/k1_defconfig >> .config
make ARCH=riscv menuconfig
make ARCH=riscv save_oedefconfig

Build and boot testing passed on k1 bananapi-f3 board.

Signed-off-by: zhangmeng <zhangmeng.kevin@linux.spacemit.com>

kevin-zhm commented:
更新了pull request

kevin-zhm commented:
/check

oervci commented:
开始测试

oervci commented:
check patch done. log: https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/check-patch/7/consoleFull

oervci commented:
Kernel build success!

oervci commented:
kunit test done. log:https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/kunit-test/9/consoleFull

oervci commented:
Lava check done! result url: https://lava.oerv.ac.cn/scheduler/job/161

wangliu-iscas commented:
/check

oervci commented:
开始测试

oervci commented:
check patch done. log: https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/check-patch/8/consoleFull

oervci commented:
Kernel build success!

oervci commented:
kunit test done. log:https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/kunit-test/10/consoleFull

oervci commented:
Lava check done! result url: https://lava.oerv.ac.cn/scheduler/job/164

woqidaideshi commented:
@kevin-zhm @sterling-teng 使用 openeuler_defconfig 编译出的内核可以正常启动 LicheePi4A。

sterling-teng commented:
> @kevin-zhm @sterling-teng 使用 openeuler_defconfig 编译出的内核可以正常启动 LicheePi4A。

OK，晚些时候我测试下milk-v pioneer。

xmzzz commented:
@kevin-zhm @sterling-teng 
Milkv Pioneer 上测试 sg2042 启动正常，
日志链接：https://gitee.com/xingmz/doc/blob/master/riscv-kernel/rvck-olk-pr-18-k1/sg2042-boot-log.txt

sterling-teng commented:
> @kevin-zhm @sterling-teng Milkv Pioneer 上测试 sg2042 启动正常， 日志链接：https://gitee.com/xingmz/doc/blob/master/riscv-kernel/rvck-olk-pr-18-k1/sg2042-boot-log.txt
OK，如果没有新的评审意见，可以合并了。
